### PR TITLE
Improve security for the api call

### PR DESCRIPTION
**Pull request by jssuzanne**
It is dangerous to be able to access all methods or classmethods without verification. In addition, some methods must not be accessible from the outside, otherwise it would be possible to be able to bypass the CRUD API

- [X] Authorizes the exposure of the method via a decorator
- [X] The notion of classmethod is defined in the controller (default is a classmethod)
- [X] The **request** can be passed in the method
- [X] The **authenticated_userid** can be passed in the method
- [X] The **resource** can be passed in the method
- [x] Permission key can be defined to validate if authenticated_userid is allowed to call this method
- [x] Validate that the method linked by a **Model.FuretUI.Menu.Call** is decorated

```
from anyblok_furetui import exposed_method


@register(Model)
class Test:

    @exposed_method(is_classmethod=False, request='request', authenticated_userid='user', resource='resource')
    def do_something(self, request=None, user=None, resource=None, **data_from_call):
        # do something
        return response

```

### CODE REVIEW

**Comment by jssuzanne** [comment]
@petrus-v i wonder if the name of the decorator is explicit enough or if we should replace it by

- call
- allow_to_be_called
- callable

or other

File: anyblok_furetui/tests/test_call_method.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+class TestCallMethod:
+    @pytest.fixture(autouse=True)
+    def transact(self, request, registry_call_method):
+        transaction = registry_call_method.begin_nested()
+        request.addfinalizer(transaction.rollback)
+        return
+
+    def call(self, webserver, call, status=200):
+        url = '/furet-ui/resource/0/model/Model.Test/call/%s' % call
```
Comment (petrus-v) [inline anyblok_furetui/tests/test_call_method.py:174]: I juste want to notice that, as far we only work with python > 3.6, I think we should avoid this old formatting operation style.

reference: https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting

> Note The formatting operations described here exhibit a variety of quirks that lead to a number of common errors (such as failing to display tuples and dictionaries correctly). Using the newer formatted string literals, the str.format() interface, or template strings may help avoid these errors. Each of these alternatives provides their own trade-offs and benefits of simplicity, flexibility, and/or extensibility.

```suggestion
        url = f'/furet-ui/resource/0/model/Model.Test/call/{call}'
```
or
```suggestion
        url = '/furet-ui/resource/0/model/Model.Test/call/{}'.format(call)
```

**Comment by petrus-v** [comment]
> @petrus-v i wonder if the name of the decorator is explicit enough or if we should replace it by
> 
> * call
> * allow_to_be_called
> * callable
> * exposed_by_call_api
> 
> or other

I like the idea to use `expose` what do you think about `furet_ui_exopsed_method` ?

**Comment by jssuzanne** [comment]
I think that furet_ui should be removed iin the name
because:

```
from anyblok_furetui import furet_ui_exposed_method
```

i prefere 

```
from anyblok_furetui import exposed_method
```

File: anyblok_furetui/furetui/system.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        # MAIBE reload UI
+
+    @exposed_method(is_classmethod=False, permission='admin')
+    def uninstall(self):
+        super(Blok, self).uninstall()
+        # MAIBE reload UI
+
+    @exposed_method(is_classmethod=False, permission='admin')
+    def upgrade(self):
+        super(Blok, self).uninstall()
```
Comment (petrus-v) [inline anyblok_furetui/furetui/system.py:28]: humm I would suggest:
```suggestion
        super(Blok, self).upgrade()
```

**Comment by petrus-v** [comment]
Looks good to me once the review: https://github.com/AnyBlok/anyblok_furetui/pull/23#discussion_r526711214 is resolved

**Pull request merged**